consensus/proto_array/src/proto_array.rs

```diff
@@ -103,6 +116,54 @@ pub struct ProtoNode {
     pub unrealized_finalized_checkpoint: Option<Checkpoint>,
 }
 
+impl Into<ProtoNode> for ProtoNodeV16 {
+    fn into(self) -> ProtoNode {
+        ProtoNode {
+            slot: self.slot,
+            state_root: self.state_root,
+            target_root: self.target_root,
+            current_epoch_shuffling_id: self.current_epoch_shuffling_id,
+            next_epoch_shuffling_id: self.next_epoch_shuffling_id,
+            root: self.root,
+            parent: self.parent,
+            justified_checkpoint: self
+                .justified_checkpoint
+                .expect("Justified checkpoint cannot be None."),
```

Comment:
These two `expect`'s aren't ideal. I could change this `impl` into a `TryInto`, (which would necessitate various downstream changes such as also using a `TryInto` for `SszContainer`).

From what I know, the `Option` was only required to facilitate a previous migration (one that support for downgrading has long since been removed) and has been `Some` ever since, so in theory it should be 100% safe.

In addition, if users do somehow run into this panic, they simply need to do a fresh checkpoint sync which will fix the issue.

Happy to hear alternatives, but in the meantime I'll look into using a `TryInto` instead.

Reply:
Using `TryInto` ended up being quite clean so these `expect`'s have been removed now.